torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,326 @@
+import collections
+import dataclasses
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import textwrap
+from typing import List, Tuple
+
+import torch
+
+
+@dataclasses.dataclass(repr=False, eq=False, frozen=True)
+class CallgrindStats(object):
+    stmt: str
+    setup: str
+    number: int
+    num_threads: int
+    built_with_debug_symbols: bool
+    baseline_inclusive_stats: List[Tuple[int, str]]
+    baseline_exclusive_stats: List[Tuple[int, str]]
+    stmt_inclusive_stats: List[Tuple[int, str]]
+    stmt_exclusive_stats: List[Tuple[int, str]]
+
+    def __repr__(self):
+        instruction_count = str(sum(c for c, _ in self.stmt_exclusive_stats))
+        baseline_count = str(sum(c for c, _ in self.baseline_exclusive_stats))
+        count_len = max(len(instruction_count), len(baseline_count))
+
+        # Pad lines after the first to align properly.
+        stmt = self.stmt.replace('\n', '\n' + ' ' * 9)
+        setup = self.setup.replace('\n', '\n' + ' ' * 9)
+        lines = [
```

Comment:
I guess this is a personal style thing, but honestly you should just use a `f""" ... """` string here :>

Reply:
The trouble with `f"""..."""` is that there's no good way to do optional lines, though `__repr__` is quite ugly (and actually even uglier than when you made this comment), so I think I'm going to give it a go.

Reply:
Well, at least in this case, `+=` the optional line at the end would be pretty simple. I agree it's more tricky if you want the optional line internally and don't want an extra whitespace, but... whatever lol.